src/Altinn.Platform/Altinn.Platform.Storage/UnitTest/TestingControllers/MessageboxInstancesControllerTests.cs

```diff
@@ -540,6 +540,108 @@ public async void Delete_ArchivedMissingRole_ReturnsForbidden()
+        /// Expected:
+        ///  There is a match for active, archived and soft deleted instances
+        /// Success:
+        ///  List of instances is returned
+        /// </summary>
+        [Fact]
+        public async void Search_FilterOnAppId_ReturnsActiveArchivedAndDeletedInstances()
+        {
+            // Arrange
+            HttpClient client = GetTestClient();
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", PrincipalUtil.GetToken(3, 1600, 3));
+
+            // Act
+            HttpResponseMessage responseMessage = await client.GetAsync($"{BasePath}/sbl/instances/search?instanceOwner.partyId=1600&appId=ttd/steffens-2020-v2");
+            string content = await responseMessage.Content.ReadAsStringAsync();
+            List<MessageBoxInstance> actualResult = JsonConvert.DeserializeObject<List<MessageBoxInstance>>(content);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, responseMessage.StatusCode);
+            Assert.Equal(1, actualResult.Count(i => i.DeleteStatus == DeleteStatusType.SoftDeleted));
+            Assert.Equal(1, actualResult.Count(i => i.ProcessCurrentTask == "FormFilling"));
+            Assert.Equal(1, actualResult.Count(i => i.ProcessCurrentTask == "Archived" && i.DeleteStatus == DeleteStatusType.Default));
+        }
+
+        /// <summary>
+        /// Scenario:
+        ///  Search instances for a given partyId and appId
+        /// Expected:
+        ///  There are two matches, but one is a hard deleted instance
+        /// Success:
+        ///  Hard deleted instances are not included in the response.
+        /// </summary>
+        [Fact]
+        public async void Search_FilterOnAppId_HardDeletedInstancesAreExcludedFromResult()
+        {
+            // Arrange
+            HttpClient client = GetTestClient();
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", PrincipalUtil.GetToken(3, 1606, 3));
+
+            int expectedCount = 1;
+
+            // Act
+            HttpResponseMessage responseMessage = await client.GetAsync($"{BasePath}/sbl/instances/search?instanceOwner.partyId=1600&appId=tdd/test-applikasjon-1&language=en");
+            string content = await responseMessage.Content.ReadAsStringAsync();
+            List<MessageBoxInstance> actualResult = JsonConvert.DeserializeObject<List<MessageBoxInstance>>(content);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, responseMessage.StatusCode);
+            Assert.Equal(expectedCount, actualResult.Count);
+        }
+
+        /// <summary>
+        /// Scenario:
+        ///  Search instances based on unknown search parameter
+        /// Expected:
+        ///  No matches are found
+        /// Success:
+        ///  Empty list is returned
+        /// </summary>
+        [Fact]
+        public async void Search_FilterOnUnknownParameter_EmptyListIsReturned()
```

Comment:
Not sure about this test.

The issue is that the implementation of the real `InstanceRepository` and the mocked counter part acts very differently in this case. In `InstanceRepositoryMock` the unknown parameter is simply ignored, but in the real implementation it'll produce an exception that is later stored in the Exception property of `InstanceQueryResponse` from `GetInstancesFromQuery`. This is a property that is never checked by the new search method.

The problem here is that an unknown parameter is terminating. The result is empty because no search was performed. Not because we didn't find anything. 

This has shown me that a search request with an unknown parameter must end up with a BadRequest.

Reply:
I was worried that a bad request would result in an "Oida, noe gikk galt" in SBL. This is the case when retrieving a single instance, but I will check the code in SBL and modity the implementation if bad requests are simply logged

Reply:
Why worry about that. Maybe that is the correct result. I don't think SBL should send unknown parameters and if they do it should fail so that it is discovered early in development. 

The alternative could be to ignore unknown parameters instead of failing, but that is a change to `InstanceRepository` and not isolated to message box. You could look into the consequences of that.

Reply:
I've added the check of valid queryParams to the mock and generating an exception if an invalid parameter is received. 
Returning bad request if it is an unknown parameter, and 500 if another exception is generated